aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -165,6 +167,11 @@ internal class RealAWSCognitoAuthPlugin(
 
     fun escapeHatch() = authEnvironment.cognitoAuthService
 
+    @InternalAmplifyApi
+    fun addToUserAgent(type: AWSCognitoAuthMetadataType, value: String) {
+        authEnvironment.cognitoAuthService.customUserAgentPairs[type.key] = value
```

Comment:
Im guessing value is just version number? If yes, then can we use `BuildConfig.VERSION_NAME` instead?

Reply:
IMO we should hardcode `value` too.

Reply:
The version information is not available here. We can add a regex to ensure the value conforms to the expected format but it must be passed in.